src/features/import/components/ImportDialog/Configure/Configuration/GenderConfigRow.tsx

```diff
@@ -0,0 +1,132 @@
+import { ArrowForward, Delete } from '@mui/icons-material';
+import {
+  Box,
+  Button,
+  FormControl,
+  IconButton,
+  InputLabel,
+  MenuItem,
+  Select,
+  Typography,
+} from '@mui/material';
+import { FC, useState } from 'react';
+
+import messageIds from 'features/import/l10n/messageIds';
+import { Msg, useMessages } from 'core/i18n';
+import { Gender, genders } from '../../../../hooks/useGenderMapping';
+
+interface GenderConfigRowProps {
+  italic?: boolean;
+  numRows: number;
+  onSelectGender: (gender: Gender) => void;
+  onDeselectGender: () => void;
+  selectedGender: Gender | null;
+  title: string;
+}
+
+const GenderConfigRow: FC<GenderConfigRowProps> = ({
+  italic,
+  numRows,
+  onSelectGender,
+  onDeselectGender,
+  selectedGender,
+  title,
+}) => {
+  const messages = useMessages(messageIds);
+  const [mapping, setMapping] = useState(false);
+
+  const showSelect = mapping || selectedGender;
+
+  return (
+    <Box display="flex" flexDirection="column">
+      <Box display="flex">
+        <Box
+          alignItems="flex-start"
+          display="flex"
+          justifyContent="space-between"
+          paddingTop={1}
+          width="50%"
+        >
+          <Box display="flex" sx={{ wordBreak: 'break-all' }} width="100%">
+            <Typography fontStyle={italic ? 'italic' : ''}>{title}</Typography>
+          </Box>
+          <ArrowForward color="secondary" sx={{ marginRight: 1 }} />
+        </Box>
+        <Box
+          alignItems="flex-start"
+          display="flex"
+          paddingRight={1}
+          width="50%"
+        >
+          {!showSelect && (
+            <Button onClick={() => setMapping(true)}>
+              <Msg
+                id={
+                  messageIds.configuration.configure.genders
+                    .showGenderSelectButton
+                }
+              />
+            </Button>
+          )}
+          {showSelect && (
+            <>
+              <FormControl fullWidth size="small">
+                <InputLabel>
+                  <Msg id={messageIds.configuration.configure.genders.gender} />
+                </InputLabel>
+                <Select
+                  label={messages.configuration.configure.genders.gender()}
+                  onChange={(event) => {
+                    const { value } = event.target;
+                    if (value === 'm' || value === 'f' || value === 'o') {
+                      onSelectGender(value);
+                    } else if (value === 'unknown') {
+                      // selecting `unknown` is like deselecting
+                      onDeselectGender();
+                    }
```

Comment:
I am not sure I agree (and I might very well be wrong!) that it should be the same to deselect as to select to import a value as `unknown` (`null`). 

What I mean is, that when selecting `unknown`, I want there to be information added to my object of mapping that says "value x should be mapped to null". But deselecting means removing the information related to "value x" in my object of mapping. This comes with the undesired side effect that if I open the gender mapping, and the first value i want to map i select should map to "unknown" - this is not recognised as me having done any mapping at all. Our validation for the mapping will believe no mapping for gender has happened, and it will be forbidden to move on to the validation step. 

What do you think?

Reply:
I might be misunderstanding, so please ignore if does not make sense.

I believe that another aspect of this decision is what happens to people for where there is no mapping in the import, but do have gender data in the database.

Not mapping anything doesn't change them in the database (the `data` object will not include `gender`), while mapping to "unknown" will overwrite whatever is already in the database (setting `data.gender` to `null`).

So IMO not mapping, and mapping to unknown is two separate actions where the former is a no-op and the latter is destructive.

Reply:
I changed the implementation to something that should now work for any of the "regular" genders, a deliberate "we want to set this to null" and a "ignore this value in the import"